cmake/CMakeBasics.cmake

```diff
@@ -5,7 +5,7 @@ set(PACKAGE_BUGREPORT "barton22@llnl.gov")
 
 set(SNLS_VERSION_MAJOR 0)
 set(SNLS_VERSION_MINOR 3)
-set(SNLS_VERSION_PATCH \"3\")
+set(SNLS_VERSION_PATCH \"0\")
```

Comment:
We'll need to update this to 4 rather than 0 for the patch number. Overall, I think this should be fine as it's more or less what we're using in what will be the eventual v0.4.0 release. I will test it with everything enabled though to verify that before merging.

Reply:
The cmake find libraries appears to work as I tested it on a HIP build of the RAJA Portability Suite and this branch of SNLS. @gberg617 I can tick the patch if you'd like or you can do it, but outside that it should be good to go.